File: lib/swift/time_conversion.swift

// convert to military time
func timeConversion(s: String) -> String {
    let time = s.components(separatedBy: ":") // ["12", "00", "00AM"]
    let hour = time[0] // "12"
    let minute = time[1] // "00"
    let second = time[2] // "00AM"

    let ampm = second.dropFirst(2) // "AM"
    
    print(ampm)

    if ampm == "AM" && hour == "12" {
        return "00:\(minute):\(second.dropLast(2))"
    } else if ampm == "PM" && hour != "12" {
        return "\(Int(hour)! + 12):\(minute):\(second.dropLast(2))"
    } else {
        return "\(hour):\(minute):\(second.dropLast(2))"
    }
}

File: lib/dart/time_conversion.dart
String timeConversion(final String s) {
  final time = s.split(':');
  final hour = int.parse(time[0]);
  final minute = int.parse(time[1]);
  final last = time[2];
  final ampm = s.substring(8, 10);

  // remove AM/PM
  final s1 = last.replaceAll(ampm, '');

  final second = int.parse(s1);

  final hour24 = ampm == 'AM' ? hour : hour + 12;
  var hour24Str = hour24.toString().padLeft(2, '0');
  if (hour == 12) {
    hour24Str = '00';
  }

  final minuteStr = minute.toString().padLeft(2, '0');
  final secondStr = second.toString().padLeft(2, '0');
  return '$hour24Str:$minuteStr:$secondStr';
}
